equinox/_filters.py

```diff
@@ -135,21 +135,45 @@ def filter(
 def partition(
     pytree: PyTree,
     filter_spec: PyTree[AxisSpec],
+    *filter_specs: PyTree[AxisSpec],
     replace: Any = None,
     is_leaf: Optional[Callable[[Any], bool]] = None,
 ) -> tuple[PyTree, PyTree]:
     """Splits a PyTree into two pieces. Equivalent to
     `filter(...), filter(..., inverse=True)`, but slightly more efficient.
 
+    More generally, provide $N$ filter specifications to split the tree into $N + 1$
+    non-overlapping partitions.
+
```

Comment:
Could you slightly extend on the documentation? Before I looked at the code, it was not obvious what multiple partitioning does. My first assumption was that you would split things with some redundancy:
```
everything_that_is_array, everything_that_is_inexact_array, rest = eqx.filter(x, eqx.is_array, eqx.is_inexact_array)
```
I feel like this is more in line with the previous partitioning logic, at least that is how my brains understand the standard partitioning. 

Actually, I really like this way to specify things, but it does indeed break backwards compatibility :( 

I would probably implement it as a separate partition function, e.g. ```eqx.multi_partition```? This way the syntax could be wonderfully concise, and compatibility will be kept intact. Anyhow, that is for Patrick to decide, I'm just passing by.

Reply:
This does also speak to another good point, how do overlapping filters work? It's not really clear which group things end up in.

The current 'only split into 2' scenario necessitates making this explicit.

Reply:
When a leaf satisfies several filters, it is contained in the first one that is satisfied, as determined by the user-provided order.